internal/resources/app_endpoint_oidc.go

```diff
@@ -0,0 +1,387 @@
+	details, err := r.getOidcProvider(ctx, organizationId, projectId, clusterId, appServiceId, appEndpointName, providerId)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error refreshing App Endpoint OIDC Provider after creation",
+			fmt.Sprintf("Could not read OIDC provider %s on App Endpoint %s ", providerId, appEndpointName)+". "+api.ParseError(err),
+		)
+		return
+	}
+	r.mapResponseToState(&plan, details)
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+}
+
+// Read fetches the OIDC provider and refreshes state.
+func (r *AppEndpointOidcProvider) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state providerschema.AppEndpointOidcProvider
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	IDs, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError("Error Reading OIDC Provider", "Could not validate state: "+err.Error())
+		return
+	}
+
+	providerId := IDs[providerschema.ProviderId]
+	if providerId == "" {
+		tflog.Info(ctx, "providerId missing; removing from state")
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	details, err := r.getOidcProvider(ctx, IDs[providerschema.OrganizationId], IDs[providerschema.ProjectId], IDs[providerschema.ClusterId], IDs[providerschema.AppServiceId], IDs[providerschema.AppEndpointName], providerId)
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "OIDC provider not found; removing from state")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("Error Reading OIDC Provider", "Could not read OIDC provider: "+errString)
+		return
+	}
+
+	// On Read, populate all fields from remote
+	r.mapResponseToState(&state, details)
+
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+// Update updates an existing OIDC provider for the App Endpoint using the Capella API.
+func (r *AppEndpointOidcProvider) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan providerschema.AppEndpointOidcProvider
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read current state to obtain provider ID
+	var state providerschema.AppEndpointOidcProvider
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := plan.OrganizationId.ValueString()
+	projectId := plan.ProjectId.ValueString()
+	clusterId := plan.ClusterId.ValueString()
+	appServiceId := plan.AppServiceId.ValueString()
+	appEndpointName := plan.AppEndpointName.ValueString()
+
+	providerId := state.ProviderId.ValueString()
+
+	if providerId == "" {
+		resp.Diagnostics.AddError("Error Updating OIDC Provider", "provider_id is missing; cannot update")
+		return
+	}
```

Comment:
since `Create` returns error if this is empty string no need for this check

Reply:
https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/396/commits/23e6765c89e2ee3e9e3c07cb267733e05126f4c2